app/scripts/migrations/101.ts

```diff
@@ -0,0 +1,36 @@
+import { cloneDeep } from 'lodash';
+
+type MetaMaskState = Record<string, any>;
+type VersionedState = {
+  meta: { version: number };
+  data: MetaMaskState;
+};
+
+export const version = 101;
+
+/**
+ * This migration removes obsolete 'isLineaMainnetReleased' PreferencesController state properties.
+ *
+ * @param originalVersionedState - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedState.meta - State metadata.
+ * @param originalVersionedState.meta.version - The current state version.
+ * @param originalVersionedState.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned of MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedState: VersionedState,
+): Promise<VersionedState> {
+  const versionedState = cloneDeep(originalVersionedState);
+  versionedState.meta.version = version;
+
+  const state = versionedState.data;
+  versionedState.data = transformState(state);
+  return versionedState;
+}
+
+function transformState(state: MetaMaskState): MetaMaskState {
+  if (state?.PreferencesController?.isLineaMainnetReleased) {
```

Comment:
Can you add this before this condition? This pattern is useful for observing any state inconsistencies through our monitoring system:

```js
 if (
    !hasProperty(state, 'PreferencesController') ||
    !isObject(state.PreferencesController)
  ) {
    captureException(
      `Migration ${version}: Invalid PreferencesController state: ${typeof state.PreferencesController}`,
    );

    return state;
  }
  ```